src/components/MaterialUI/hocs/withDeferredPopper.js

```diff
@@ -0,0 +1,127 @@
+import React, { useState } from "react";
+import Popper from '@material-ui/core/Popper';
+import ClickAwayListener from '@material-ui/core/ClickAwayListener';
+import {
+  isString,
+  isObject,
+  isStringNullOrWhitespace,
+  isReactComponent,
+} from "../../../utils/propertyValidator";
+import { makeStyles } from "@material-ui/core/styles";
+
+const useStyles = makeStyles(theme => ({
+  popper: {
+    zIndex: 1,
+    '&[x-placement*="bottom"] $arrow': {
+      top: 0,
+      left: "50%",
+      transform: "translateX(-50%) rotate(45deg)",
+      marginTop: '-0.25em',
+      width: '0.4em',
+      height: '0.4em',
+      borderColor: `${theme.palette.grey.icon} transparent transparent ${theme.palette.grey.icon}`,
+    },
+    '&[x-placement*="top"] $arrow': {
+      bottom: 0,
+      left: "50%",
+      transform: "translateX(-50%) rotate(45deg)",
+      marginBottom: '-0.25em',
+      width: '0.4em',
+      height: '0.4em',
+      borderColor: `transparent ${theme.palette.grey.icon} ${theme.palette.grey.icon} transparent`,
+    },
+    '&[x-placement*="right"] $arrow': {
+      left: 0,
+      top: "50%",
+      marginLeft: '-0.25em',
+      transform: "translateY(-50%) rotate(45deg)",
+      width: '0.4em',
+      height: '0.4em',
+      borderColor: `transparent transparent ${theme.palette.grey.icon} ${theme.palette.grey.icon}`,
+    },
+    '&[x-placement*="left"] $arrow': {
+      right: 0,
+      top: "50%",
+      marginRight: '-0.25em',
+      transform: "translateY(-50%) rotate(45deg)",
+      width: '0.4em',
+      height: '0.4em',
+      borderColor: `${theme.palette.grey.icon} ${theme.palette.grey.icon} transparent transparent`,
+    },
+  },
+  arrow: {
+    position: "absolute",
+    border: "1px solid",
+    backgroundColor: theme.palette.background.paper,
+  }
+}));
+
+export const Arrow = () => {
```

Comment:
You can also set classes as Props of Arrow if you want, up to you. This is what I did for the Table to avoid to use the hook at every level.

Reply:
agree